From 5cbb0fad4571bab49953261c1336cb70a078417c Mon Sep 17 00:00:00 2001
From: Prarit Bhargava <prarit@redhat.com>
Date: Thu, 28 Aug 2014 14:22:54 -0400
Subject: [cpufreq] conservative: set requested_freq to policy max when it is over policy max

Message-id: <1409235879-1283-81-git-send-email-prarit@redhat.com>
Patchwork-id: 89533
O-Subject: [RHEL7.1 PATCH BZ 1134639 080/185] cpufreq: conservative: set requested_freq to policy max when it is over policy max
Bugzilla: 1134369
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>
RH-Acked-by: Mark Langsdorf <mlangsdo@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1134639

commit 6d7bcb1464a89181ddc4b4584ad6e0c7566ae31b
Author: Xiaoguang Chen <chenxg@marvell.com>
Date:   Fri Nov 8 13:23:52 2013 +0800

    cpufreq: conservative: set requested_freq to policy max when it is over policy max

    When requested_freq is over policy->max, set it to policy->max.
    This can help to speed up decreasing frequency.

    Signed-off-by: Xiaoguang Chen <chenxg@marvell.com>
    Acked-by: Viresh Kumar <viresh.kumar@linaro.org>
    Signed-off-by: Rafael J. Wysocki <rafael.j.wysocki@intel.com>

Cc: Lenny Szubowicz <lszubowi@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/cpufreq/cpufreq_conservative.c b/drivers/cpufreq/cpufreq_conservative.c
index 27c332a..57ada02 100644
--- a/drivers/cpufreq/cpufreq_conservative.c
+++ b/drivers/cpufreq/cpufreq_conservative.c
@@ -68,6 +68,9 @@ static void cs_check_cpu(int cpu, unsigned int load)
 
   dbs_info->requested_freq += get_freq_target(cs_tuners, policy);
 
+  if (dbs_info->requested_freq > policy->max)
+   dbs_info->requested_freq = policy->max;
+
   __cpufreq_driver_target(policy, dbs_info->requested_freq,
    CPUFREQ_RELATION_H);
   return;
-- 
1.7.1